packages/pwa-kit-dev/src/utils/script-utils.ts

```diff
@@ -0,0 +1,312 @@
+                    const archive = archiver('tar')
+
+                    archive.pipe(output)
+
+                    // See https://archiverjs.com/docs/global.html#TarEntryData
+                    const newRoot = path.join(projectSlug, 'bld', '')
+                    archive.directory(buildDirectory, '', (entry) => {
+                        if (entry.stats.isFile()) {
+                            filesInArchive.push(entry.name)
+                        }
+                        entry.prefix = newRoot
+                        return entry
+                    })
+
+                    archive.on('error', reject)
+                    output.on('finish', resolve)
+
+                    archive.finalize()
+                })
+        )
+        .then(() => readFile(destination))
+        .then((data) => {
+            const encoding = 'base64'
+            return {
+                message,
+                encoding,
+                data: data.toString(encoding),
+                ssr_parameters,
+                ssr_only: filesInArchive.filter(glob(ssr_only)),
+                ssr_shared: filesInArchive.filter(glob(ssr_shared))
+            }
+        })
+        .finally(() => rmdir(tmpDir, {recursive: true}))
+}
+
+type MatchFn = (a: string) => boolean
+
+export const glob = (patterns?: string[]): MatchFn => {
+    // The patterns can include negations, so matching is done against all
+    // the patterns. A match is true if a given path matches any pattern and
+    // does not match any negating patterns.
+
+    const allPatterns = (patterns || [])
+        .map((pattern) => new Minimatch(pattern, {nocomment: true}))
+        .filter((pattern) => !pattern.empty)
+    const positivePatterns = allPatterns.filter((pattern) => !pattern.negate)
+    const negativePatterns = allPatterns.filter((pattern) => pattern.negate)
+
+    return (path: string) => {
+        if (path) {
+            const positive = positivePatterns.some((pattern) => pattern.match(path))
+            const negative = negativePatterns.some((pattern) => !pattern.match(path))
+            return positive && !negative
+        }
+        return false
+    }
+}
+
+export const findHomeDir = (platform: string = process.platform) =>
+    platform === 'win32' ? process.env.USERPROFILE : process.env.HOME
+
+export const getCredentialsFile = (
+    cloudOrigin: string,
+    credentialsFile?: string,
+    doFindHomeDir: () => string = findHomeDir
+): string => {
+    if (credentialsFile) {
+        return credentialsFile
+    } else {
+        const url = new URL(cloudOrigin)
+        const host = url.host
+        const suffix = host === 'cloud.mobify.com' ? '' : `--${host}`
+        return path.join(doFindHomeDir(), `.mobify${suffix}`)
+    }
+}
+
+export const readCredentials = async (filepath: string): Promise<Credentials> => {
+    try {
+        const content = await readFile(filepath)
+        const data = JSON.parse(content.toString('utf-8'))
```

Comment:
```suggestion
        const content = await readJson(filepath)
```
`fs-extra` is available.